dask/array/overlap.py

```diff
@@ -474,21 +477,48 @@ def overlap(x, depth, boundary):
            [100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100],
            [100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100]])
     """
+    from .routines import aligned_coarsen_chunks
+
+    def coerce_depth_in_chunk(depth_values, chunks, tries=None):
+        """Ensure that chunk size is larger than depth"""
+        new_chunks = [c for c in chunks]
+        i = 0
+        if tries is None:
+            tries = {i: 0 for i in range(len(chunks))}
+        for d, c in zip(depth_values, chunks):
+            maxd = max(d) if isinstance(d, tuple) else d
+            if maxd > min(c):
+                if maxd > c[-1] and len(c) >= 2:
+                    # if the last chunk is the culprit, add it to prior
+                    new_chunks[i] = (*c[:-2], c[-2] + c[-1])
+                else:
+                    # otherwise coarsen all chunks
+                    new_chunks[i] = aligned_coarsen_chunks(chunks=c, multiple=maxd)
```

Comment:
I'm not sure that this function is what we want here.  We don't care about chunks being a multiple of a number, we care about them being at least a certain size